gef-ui/server/controllers/application-abandon/index.test.js

```diff
@@ -0,0 +1,177 @@
+import {
+  confirmAbandonApplication,
+  abandonApplication,
+} from './index';
+import * as api from '../../services/api';
+
+const MockResponse = () => {
+  const res = {};
+  res.redirect = jest.fn();
+  res.render = jest.fn();
+  return res;
+};
+
+const MockRequest = () => {
+  const req = {};
+  req.params = {};
+  req.query = {};
+  req.params.applicationId = '123';
+  req.session = {
+    user: {
+      bank: { id: 'BANKID' },
+      roles: ['MAKER'],
+    },
+  };
+  return req;
+};
+
+const MockApplicationResponse = () => {
+  const res = {};
+  res._id = '1234';
+  res.exporterId = '123';
+  res.coverTermsId = '123';
+  res.bankId = 'BANKID';
+  res.bankInternalRefName = 'My test';
+  res.status = 'Draft';
+  return res;
+};
+
+const MockExporterResponse = () => {
+  const res = {};
+  res.details = {};
+  res.status = 'IN_PROGRESS';
+  res.validation = {};
+  res.details.companiesHouseRegistrationNumber = 'tedsi';
+  res.validation.required = [];
+  return res;
+};
+
+const MockCoverTermsResponse = () => {
+  const res = {};
+  res.status = 'NOT_STARTED';
+  res.details = {};
+  res.validation = {};
+  res.validation.required = [];
+  res.data = [];
+  return res;
+};
+
+const MockEligibilityCriteriaResponse = () => ({
+  terms: [
+    {
+      id: 'coverStart',
+      htmlText: '<p>Some eligibility criteria</p>',
+      errMsg: '12. Select some eligibilty',
+    },
+  ],
+});
+
+const MockFacilityResponse = () => {
+  const res = {};
+  res.status = 'IN_PROGRESS';
+  res.data = [];
+  return res;
+};
+
```

Comment:
Not for this PR, seems to be a lot of duplication of the mock responses, would be nice to move into one shared file

Reply:
yeah I will refactor that later when we have time